### PR TITLE
APP-528 - Data Enrichment - add support to the PHP enforcer

### PR DESCRIPTION
**Pull request by liornaar-px**
added data enrichment

### CODE REVIEW

File: src/Perimeterx.php [hunk 1]
```diff
@@ -59,9 +59,7 @@ public static function Instance(array $pxConfig = [])
 
     private function __construct(array $pxConfig = [])
     {
-        if (!function_exists("mcrypt_encrypt")) {
-            throw new PerimeterxException(PerimeterxException::$MCRYPT_MISSING);
-        }
+
```
Comment (bendpx) [inline src/Perimeterx.php]: why did we remove the test if mcrypt exists?

File: src/PerimeterxDataEnrichment.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<?php
+namespace Perimeterx;
+
+class PerimeterxDataEnrichment {
+
+public static function processDataEnrichment($pxCtx, $pxConfig) {
```
Comment (bendpx) [inline src/PerimeterxDataEnrichment.php]: a unit test for this part of the code would be nice

File: src/PerimeterxS2SValidator.php [hunk 3]
```diff
@@ -131,6 +131,11 @@ public function verify()
             $this->pxCtx->setUuid($response->uuid);
             $this->pxCtx->setBlockAction($response->action);
             $this->pxCtx->setResponseBlockAction($response->action);
+            if(isset($response->data_enrichment)) {
+                $this->pxConfig['logger']->debug("{$response->data_enrichment->timestamp}");
```
Comment (bendpx) [inline src/PerimeterxS2SValidator.php]: why are we logging the timestamp?

File: src/Perimeterx.php (hunk 1 above)
Comment (pxjohnny) [reply src/Perimeterx.php]: mcrypt is no longer in use. there is a different Jira to remove all instances that reference it.

File: src/PerimeterxDataEnrichment.php (hunk 2 above)
Comment (liornaar-px) [reply src/PerimeterxDataEnrichment.php]: will add

File: src/PerimeterxS2SValidator.php (hunk 3 above)
Comment (liornaar-px) [reply src/PerimeterxS2SValidator.php]: was left there by mistake, thanks for the catch

**Pull request merged**